Mathlib/Analysis/Complex/Hadamard.lean

```diff
@@ -0,0 +1,487 @@
+      apply DiffContOnCl.mono _ _
+      · use strip 0 1
+      · exact diffContOnCl_F' _ _ hd
+      · exact inter_subset_left _ _
+
+    apply norm_le_of_forall_mem_frontier_norm_le bdd_strip_is_bounded hd_subset _ _
+
+    -- Frontier bounded by one
+    · intros z hfz
+      rw [bddStrip, ← Set.reProdIm, frontier_reProdIm] at hfz
+      cases' hfz with hfz1 hfz2
+      · cases' hfz1 with hfz_re hfz_im
+        rw [frontier_Ioo] at hfz_im
+        rw [closure_Ioo (zero_ne_one' ℝ)] at hfz_re
+        apply h_h; apply h_w_pos z hfz_re
+        -- `|T| = |z.im|`
+        apply le_of_eq
+        rwa [eq_comm, abs_eq (le_trans (zero_le_one' ℝ) (le_max_right B 1)),
+         ← mem_singleton_iff, or_comm, ← mem_insert_iff, ← mem_preimage]
+        -- `-T < T`
+        exact neg_lt_self_iff.mpr (lt_of_lt_of_le (zero_lt_one' ℝ) (le_max_right B 1))
+      · replace hfz_re := mem_of_mem_inter_left hfz2
+        rw [frontier_Ioo (zero_lt_one' ℝ)] at hfz_re
+        exact edges_le_one z f n hB hfz_re
+    -- Local goal: `z ∈ closure (bddStrip 0 1 T)` with `h : |z.im| ≤ T` and `hz : z ∈ strip 0 1`.
+    · rw [bddStrip, ← Set.reProdIm, closure_reProdIm, closure_Ioo (zero_ne_one' ℝ), closure_Ioo,
+      Set.reProdIm, mem_inter_iff]
+      refine ⟨hz, ?_⟩
+      rwa [mem_preimage, mem_Icc, ← abs_le]
+      apply ne_of_lt
+      exact neg_lt_self_iff.mpr (lt_of_lt_of_le (zero_lt_one' ℝ) (le_max_right B 1))
+
+  -- Now : `T < |z.im|`.
+  · simp only [not_le] at h; apply h_h; exact h_w_pos z hz (le_of_lt h)
+
+--Proof that `F'` Tendsto `F`
+lemma tendsto_F'_atTop_F (f : ℂ → ℂ) (z : ℂ) :
+    Tendsto (fun n : ℕ ↦ F' n f z ) atTop (nhds (F f z)) :=
+  have mul_const : Tendsto (fun n : ℕ ↦ (z^2-1) * (n : ℝ)⁻¹) atTop (nhds 0) := by
+    simpa only [mul_zero]
+      using tendsto_const_nhds.mul (tendsto_algebraMap_inverse_atTop_nhds_0_nat ℂ)
+
+  have comp_exp : Tendsto (fun n : ℕ ↦ exp ( (z^2-1) * (n : ℝ)⁻¹)) atTop (nhds 1) := by
+    simpa only [exp_zero]
+      using  (Continuous.tendsto continuous_exp 0).comp mul_const
+
+  by simpa only [mul_one]
+    using tendsto_const_nhds.mul comp_exp
+
+-- Proof that `abs F'` Tendsto `abs F`
+lemma tendsto_F'_atTop_F_abs (f : ℂ → ℂ) (z : ℂ) :
+    Tendsto (fun n : ℕ ↦ (abs ∘ (F' n f)) z ) atTop (nhds ((abs ∘ (F f)) z)) :=
+  (Continuous.tendsto continuous_abs (F f z)).comp (tendsto_F'_atTop_F f z)
+
+
+-- We are now ready to combine of `Hadamard_sequence` with `tendsto_F'_atTop_F_abs`:
+
+/--
+**Hadamard three-line theorem** on `[0,1]`: If `f` is a bounded function, continuous on `[0,1]`
+and differentiable on `(0,1)`, then for `M(x) := sup ((abs ∘ f) '' (re ⁻¹' {x}))` we have that
+`∀ z ∈ [0,1]` the inequality `|f(z)| ≤ |M(0)^(1-z)| * |M(1)^z|` holds.
+-/
+theorem abs_le_interp_on_closedStrip (f : ℂ → ℂ) (hd : DiffContOnCl ℂ f (strip 0 1))
+    (hB : BddAbove ((abs ∘ f) '' (closedStrip 0 1))) (z : ℂ) (hz : z ∈ closedStrip 0 1) :
+    (abs (f z • invInterpStrip f z)) ≤ 1 := by
```

Comment:
The statement does not really match the docstring: you have an upper bound of a quotient by one here. I think it would be worth it getting the statement in the form `|f z| \le ...`.